solutions/usaco-1138.mdx

```diff
@@ -0,0 +1,102 @@
+---
+id: usaco-1138
+source: USACO Gold 2021 Open
+title: Portals
+author: Maggie Liu
+---
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+using namespace std;
+
+struct Edge
+{
+	int from, to, cost;
+};
+const int MAX_N = 100000;
+int parent[MAX_N * 2];
+int comp_size[MAX_N * 2];
+
+// BeginCodeSnip{Standard DSU operations}
+void init(int n)
+{
+	for (int i = 0; i < n; i++)
+	{
+		parent[i] = i;
+		comp_size[i] = 1;
+	}
+}
+
+int find(int a)
+{
+	if (a == parent[a])
+	{
+		return a;
+	}
+	return parent[a] = find(parent[a]);
```

Comment:
```suggestion
	return a == parent[a] ? a : (parent[a] = find(parent[a]));
```

Reply:
ehh this looks concise but i'm pretty sure it decreases readability. what do other people think?

Reply:
I think it's fine as is. If you really want to be concise, you should use one array for parents and sizes like I do here:

https://github.com/bqi343/USACO/blob/4aa96cd195a770c3a7f8977441020036d84b4f24/Implementations/content/graphs%20(12)/DSU/DSU%20(7.6).h

Reply:
removes 4 lines, and i don't think any people in gold don't know about the ternary expression

Reply:
i still think the original is more readable...